trinity/plugins/builtin/console/plugin.py

```diff
@@ -0,0 +1,91 @@
+from argparse import (
+    Action,
+    ArgumentParser,
+)
+import sys
+
+from trinity.config import (
+    ChainConfig
+)
+from trinity.extensibility import (
+    BaseEvent,
+    BasePlugin,
+    PluginContext,
+    PluginProcess,
+)
+from trinity.extensibility.events import (
+    NetworkProcessReadyEvent,
+    TrinityStartupEvent,
+)
+
+from trinity.plugins.builtin.console.console import (
+    console
+)
+
+
+class ConsolePlugin(BasePlugin):
+
+    def __init__(self) -> None:
+        self.is_attach = False
+        self.is_console = False
+        self.is_vanilla = False
```

Comment:
I think we should drop the vanilla/ipython distinction and just make them separate plugins.  We can autodetect which plugin to load by default based on whether ipython is available or not.